docs/proposals/edgemesh-design.md

```diff
@@ -76,6 +76,149 @@ Add two endpoints definition:
 |ServiceBus|Target|Request Forward|
 |Router|Source|Resutst Forward, Service Discovery, Load Balance|
 
+<img src="../images/proposals/router-design.png">
+
+* R1: EdgeController watches on k8s apiserver for Services/Endpoints.
+* R2: EC get the service definition and create router rule definition and save in DB(sqlite).
+* R3: Router gets the rule definition from DB and initializes the listeners/handlers for source and target.
+* R4: Users/Applications create a request.
+* R5: Router receives the request and forwards it to the target based on rule.
+
+Since Router fetches rules from DB, in later versions it can be started as a different process than edge-controller. Also it enables router to retrieve all the rules after process restarts.
```

Comment:
How router be aware of the Kubernetes Service and Endpoints update from SQLite? I assume there need to be a cache to serve customer request. If the cache missed, router will try to retrieve data from SQLite?

Reply:
I plan to have timely syncing of rules from DB something like below
```go
// ListRules query rule list
func ListRules(projectID string, updateTime time.Time, offset int64) (rs []model.Rule, err error) {
	ormer := newOrmer()
	qs := ormer.QueryTable("rule")
	if projectID != "" {
		qs = qs.Filter("project_id", projectID)
	}
	// regardless of update time, use time.Time{}
	if !updateTime.IsZero() {
		qs = qs.Filter("updated_at__gte", updateTime)
	}
	_, err = qs.Limit(limit, offset).RelatedSel().OrderBy("updated_at").All(&rs)

	return
}
```

Reply:
Why don't consider etcd watch?

Reply:
Done, Thanks